Forwarded: not-needed
Origin: https://github.com/mroonga/mroonga/issues/298#issuecomment-1030815927
Bug: https://github.com/mroonga/mroonga/issues/298
From: Sutou Kouhei <kou@clear-code.com>
Date: Sat, 5 Feb 2022 11:05:39 +0900
Subject: [PATCH] cmake: add support for reproducible buildS
 .
 We should use relative path not absolute path.
 We can use target without breaking reproducibility.
--- a/storage/mroonga/CMakeLists.txt
+++ b/storage/mroonga/CMakeLists.txt
@@ -209,7 +209,7 @@ set(MYSQL_INCLUDE_DIRS
 
 if(MRN_BUNDLED)
   set(MYSQL_PLUGIN_DIR "${INSTALL_PLUGINDIR}")
-  set(MYSQL_SERVICES_LIB_DIR "${MYSQL_BUILD_DIR}/libservices")
+  set(MYSQL_SERVICES_LIB_DIR)
   set(MYSQL_CFLAGS "${CMAKE_C_FLAGS}")
   set(MYSQL_VERSION "${MYSQL_BASE_VERSION}")
 else()
@@ -248,15 +248,11 @@ endif()
 
 if(MRN_GROONGA_BUNDLED)
   set(GROONGA_INCLUDE_DIRS "${MRN_BUNDLED_GROONGA_DIR}/include")
-  set(GROONGA_LIBRARY_DIRS "${MRN_BUNDLED_GROONGA_DIR}/lib")
-  set(GROONGA_LIBRARIES "libgroonga")
+  set(GROONGA_LIBRARY "libgroonga")
 
-  set(MRN_LIBRARY_DIRS ${GROONGA_LIBRARY_DIRS})
-  set(MRN_LIBRARIES ${GROONGA_LIBRARIES})
+  set(MRN_LIBRARY_DIRS)
+  set(MRN_LIBRARIES ${GROONGA_LIBRARY})
   if(MRN_GROONGA_NORMALIZER_MYSQL_EMBED)
-    set(MRN_LIBRARY_DIRS
-      ${MRN_LIBRARY_DIRS}
-      "${MRN_BUNDLED_GROONGA_NORMALIZER_MYSQL_DIR}/normalizers")
     set(MRN_LIBRARIES ${MRN_LIBRARIES} mysql_normalizer)
   endif()
 else()
